src/MassTransit/Logging/Diagnostics/LogContextActivityExtensions.cs

```diff
@@ -17,7 +17,9 @@ public static class LogContextActivityExtensions
             params (string Key, object? Value)[] tags)
             where T : class
         {
-            var activity = Cached.Source.Value.CreateActivity(transportContext.ActivityName, ActivityKind.Producer);
+            var parentActivityContext = GetParentActivityContext(context.Headers);
```

Comment:
Okay, so, if Activity.Current != null, this shouldn't be happening as there is already a parent context present (likely from the consumed message, saga, etc.).

Reply:
Okay, so maybe something like?
```csharp
var parentActivityContext = Activity.Current?.Context ?? GetParentActivityContext(context.Headers);
```

Or should this be baked in `GetParentActivityContext`? Note that this would affect every other activity.

Reply:
```
var parentActivityContext = Activity.Current?.Context ?? GetParentActivityContext(context.Headers);
```

That seems like the right way to resolve it. The receive activity expects to pull it from the headers every time, so it shouldn't check current. I think this line in those two methods would be sufficient, but subsequent activities after the receiveactivity are significant.

Reply:
9de76fb

Reply:
Actually broke some tests, didn't realize there were telemetry tests (good). Looking into it.

Reply:
Honestly having some trouble figuring out why `baggage` is not being propagated with my change. Still, I changed it up to only use the `Activity.Current`'s if the header's activity context is not present. 14157d5

Any idea why? Maybe I'm missing something about the Activity API.

Reply:
So did you figure it out? I see the latest build is green.